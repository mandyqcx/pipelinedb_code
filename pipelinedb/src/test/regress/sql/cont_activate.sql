CREATE FOREIGN TABLE cont_activate_s (x int) SERVER pipelinedb;

CREATE VIEW cont_activate_v0 AS SELECT count(*) FROM cont_activate_s;
CREATE VIEW cont_activate_v1 AS SELECT count(*) FROM cont_activate_s;

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;

SELECT * FROM cont_activate_v0;
SELECT * FROM cont_activate_v1;

SELECT pipelinedb.deactivate('cont_activate_v0');

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;

SELECT * FROM cont_activate_v0;
SELECT * FROM cont_activate_v1;

SELECT pipelinedb.deactivate('cont_activate_v1');

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;

SELECT * FROM cont_activate_v0;
SELECT * FROM cont_activate_v1;

SELECT pipelinedb.activate('cont_activate_v0');

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;

SELECT * FROM cont_activate_v0;
SELECT * FROM cont_activate_v1;

SELECT pipelinedb.activate('cont_activate_v1');

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;

SELECT * FROM cont_activate_v0;
SELECT * FROM cont_activate_v1;

DROP VIEW cont_activate_v0;
DROP VIEW cont_activate_v1;

DROP FOREIGN TABLE cont_activate_s CASCADE;
