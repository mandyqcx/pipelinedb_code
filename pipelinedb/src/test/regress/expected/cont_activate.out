CREATE FOREIGN TABLE cont_activate_s (x int) SERVER pipelinedb;
CREATE VIEW cont_activate_v0 AS SELECT count(*) FROM cont_activate_s;
CREATE VIEW cont_activate_v1 AS SELECT count(*) FROM cont_activate_s;
INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;
SELECT * FROM cont_activate_v0;
 count 
-------
   100
(1 row)

SELECT * FROM cont_activate_v1;
 count 
-------
   100
(1 row)

SELECT pipelinedb.deactivate('cont_activate_v0');
 deactivate 
------------
 t
(1 row)

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;
SELECT * FROM cont_activate_v0;
 count 
-------
   100
(1 row)

SELECT * FROM cont_activate_v1;
 count 
-------
   200
(1 row)

SELECT pipelinedb.deactivate('cont_activate_v1');
 deactivate 
------------
 t
(1 row)

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;
SELECT * FROM cont_activate_v0;
 count 
-------
   100
(1 row)

SELECT * FROM cont_activate_v1;
 count 
-------
   200
(1 row)

SELECT pipelinedb.activate('cont_activate_v0');
 activate 
----------
 t
(1 row)

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;
SELECT * FROM cont_activate_v0;
 count 
-------
   200
(1 row)

SELECT * FROM cont_activate_v1;
 count 
-------
   200
(1 row)

SELECT pipelinedb.activate('cont_activate_v1');
 activate 
----------
 t
(1 row)

INSERT INTO cont_activate_s (x) SELECT generate_series(1, 100) AS x;
SELECT * FROM cont_activate_v0;
 count 
-------
   300
(1 row)

SELECT * FROM cont_activate_v1;
 count 
-------
   300
(1 row)

DROP VIEW cont_activate_v0;
DROP VIEW cont_activate_v1;
DROP FOREIGN TABLE cont_activate_s CASCADE;
